### ISSUE TITLE
blazor.server.js Uncaught type error remove child of null

### ISSUE THREAD

**Issue by bizcad** [dotnet/aspnetcore#27656]
### Describe the bug
In a Blazor Server app
I add a javascript charting component to a razor page.  I add new razor page and javascript file.  I add a data service. I call javascript from within the razor using IJSRuntime.  The chart displays but so does an error message.  The error appears to be in blazor.server.js because the page freezes and SignalR disconnects from the server.

#### Three error messages show up in the javascript console.
Error: There was an error applying batch 9.
Uncaught (in promise) TypeError: Cannot remove property 'removeChild' of null  (with stack trace)
System.AggregationException: One or more errors occurred.

See screenshot for details in project and repo at wwwroot\images\console.png  

### To Reproduce

Here is a [link ](https://github.com/bizcad/Vega-LiteBlazor) to the repo  https://github.com/bizcad/Vega-LiteBlazor

The project is based on a stock new blazor server project.  A page is added along with a data server, a data object, a javascript file, an image of the error.  I referenced the javascript file, three cdn links to vega in _Host.chtml.  There is also a <script> tag for testing based on the docs example.  The data server is registered in Startup.cs.  The page calls the data server and pushes the data as json into the chart control in the @code on the page.  The javascript page parses the json and replaces the data.values node before embedding it on the page.

Steps to reproduce
============
1.  Clone the project
2.  Restore nuget package (Newtonsoft.Json)
3.  Compile
4.  Start with IISExpress. 
5.  Click the new nav item VegaData.
6.  Click the Embed Visualization button.
7.  Open Devtools to examine the error

You can examine a png of the console output (console.png) in the wwwroot\images folder

#### Error shown on page
An unhandled exception has occurred. See browser dev tools for details. Reload
#### Error shown in dev tools.
Opening dev tools, three red error messages show up in the javascript console.
Error: There was an error applying batch 9.
Uncaught (in promise) TypeError: Cannot remove property 'removeChild' of null  (with stack trace)
System.AggregationException: One or more errors occurred. 

### Further technical details
- ASP.NET Core version
5.0.0-rc.2.20475.5
- Include the output of `dotnet --info`
.NET SDK (reflecting any global.json):
 Version:   5.0.100-rc.2.20479.15
 Commit:    da7dfa8840

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.19041
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\5.0.100-rc.2.20479.15\

Host (useful for support):
  Version: 5.0.0-rc.2.20475.5
  Commit:  c5a3f49c88

.NET SDKs installed:
  2.1.802 [C:\Program Files\dotnet\sdk]
  3.1.403 [C:\Program Files\dotnet\sdk]
  5.0.100-preview.8.20417.9 [C:\Program Files\dotnet\sdk]
  5.0.100-rc.2.20479.15 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.All 2.1.2 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.1.13 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.1.21 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.1.23 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.2 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.1.13 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.1.21 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.1.23 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.3 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.7 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.9 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.0-preview.8.20414.8 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.0-rc.1.20451.17 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.0-rc.2.20475.17 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.1.13 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.21 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.23 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.7 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.9 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.0-preview.8.20407.11 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.0-rc.2.20475.5 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 3.1.7 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 3.1.9 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 5.0.0-preview.8.20411.6 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 5.0.0-rc.2.20475.6 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
- The IDE (VS / VS Code/ VS4Mac) you're running on, and it's version
Microsoft Visual Studio Community 2019 Preview
Version 16.8.0 Preview 6.0
![Console](https://user-images.githubusercontent.com/995059/98601031-4fd79f00-2293-11eb-91aa-8692665a7497.png)

**Comment by pranavkm** [comment]
FYI @SteveSandersonMS

**Comment by SteveSandersonMS** [comment]
Thanks for reporting this, @bizcad!

With the repro app you've posted, there's a *lot* going on in there, so it would take quite a lot of debugging to make sense of whether the problem is in your code or in the framework. If you're looking for a quicker resolution to this issue, it would be great if you could supply a really minimal repro. Can you reduce the repro steps to something tiny like "Add these 10-20 lines of code to the default project template"?

If you can do that we'll almost definitely be able to answer straight away about whether it's an issue in your code or a framework bug.

**Comment by bizcad** [comment]
@SteveSandersonMS @pranavkm I have discovered the problem and changed the [posted repo app](https://github.com/bizcad/Vega-LiteBlazor).  Instead of trying to assign the data to the Vega-Lite spec.data from a Json array, I cleared the spec.data array and pushed each member of the parameter array into the spec array.  That cleared the problem.  **Close the ticket** with my apologies for taking your time.
